### PR TITLE
[SETUP] Wait for ShowStatusMessageThread to terminate

### PR DESCRIPTION
**Pull request by tkreuzer**
Previously CommonInstall() could return while the thread was still running and the caller would then free syssetup.dll.

### CODE REVIEW

**Comment by HBelusca** [comment]
Hmm.. what is that other thread you are mentioning? It seems that InstallReactOS is just mono-thread; it only creates a thread for managing the Shift-F10 hotkey for starting cmd.exe ...

**Comment by tkreuzer** [comment]
@HBelusca Exactly.

**Comment by ThFabba** [comment]
It seems cleaner to do a `WaitForSingleObject` on `hHotkeyThread` instead. The only reason I didn't do that originally was that it seemed unnecessary since the app is terminating anyway -- if it can cause a crash due to the lib unload, it seems worth it to wait.

**Comment by HBelusca** [comment]
Then one also needs to stop manually/kill the hHotkeyThread.

File: dll/win32/syssetup/install.c [hunk 1]
```diff
@@ -682,49 +685,54 @@ CommonInstall(VOID)
...
+
+    if (bResult == FALSE)
+    {
+        SetupCloseInfFile(hSysSetupInf);
+    }
+
+    if (hThread)
+    {
+        bEnd = TRUE;
+        WaitForSingleObject(hThread, INFINITE);
```
Comment (learn-more) [inline dll/win32/syssetup/install.c]: Maybe something like `PostThreadMessage(WM_NULL, .....` to fire the message loop?

File: dll/win32/syssetup/install.c [hunk 2]
```diff
@@ -682,49 +684,54 @@ CommonInstall(VOID)
...
     if (CMP_WaitNoPendingInstallEvents(INFINITE) != WAIT_OBJECT_0)
     {
         FatalError("CMP_WaitNoPendingInstallEvents() failed!\n");
-        goto error;
+        goto Exit;
     }
 
-    EndDialog(hWnd, 0);
-    return TRUE;
+    bResult= TRUE;
```
Comment (HBelusca) [inline dll/win32/syssetup/install.c]: space before the equal sign.

**Pull request merged**